test/fika/parser_test.exs

```diff
@@ -161,20 +187,28 @@ defmodule Fika.ParserTest do
       ]
     end
 
-    test "with return type" do
-      str = """
-      fn foo : Int do
-        123
-      end
-      """
+    Enum.each([
+      "Int",
+      ":ok"
+    ],
+    fn declared_type ->
+      test "with return type #{declared_type}" do
+        declared_type = unquote(declared_type)
 
-      {:ok, result, _rest, _context, _line, _byte_offset} = Parser.function_def(str)
+        str = """
+        fn foo : #{declared_type} do
+          123
+        end
+        """
 
-      assert result == [
-        {:function, [position: {3, 22, 25}],
-          {:foo, [], {:type, {1, 0, 12}, "Int"}, [{:integer, {2, 16, 21}, 123}]}}
-      ]
-    end
+        {:ok, result, _rest, _context, _line, _byte_offset} = Parser.function_def(str)
+
+        assert result == [
+                 {:function, [position: {3, 22, 25}],
+                 {:foo, [], {:type, {1, 0, 12}, declared_type}, [{:integer, {2, 16, 21}, 123}]}}
+              ]
+      end
+    end)
```

Comment:
I'm thinking we don't have to add a test here because we didn't change the way we infer function definitions. As long as atom expressions are inferred properly, function definitions should work. It's nice that you thought of this though! Side note: it'd be nice to get rid of Enum.each to DRY our tests. Separate tests are preferred in Fika even if we have to duplicate some code.

Reply:
> I'm thinking we don't have to add a test here because we didn't change the way we infer function definitions. As long as atom expressions are inferred properly, function definitions should work.

I would prefer to keep this test, because if [this change at parser.ex:337](https://github.com/fika-lang/fika/pull/16/files#diff-6d816435468fa63d736e9f9dd8e1f0e4R218) is removed, this test breaks.
However, since the type-specific tests also break, it wouldn't be a problem. What do you think, @emilsoman?

> It's nice that you thought of this though! Side note: it'd be nice to get rid of Enum.each to DRY our tests. Separate tests are preferred in Fika even if we have to duplicate some code.

Regardless of the answer to the above question, I'll make sure to separate the tests accordingly!

Reply:
Got it, let's keep the test 👍